adb/adb_commands.py

```diff
@@ -154,6 +154,7 @@ def Push(self, source_file, device_filename, mtime='0', timeout_ms=None):
         self.handle, destination=b'sync:', timeout_ms=timeout_ms)
     self.filesync_handler.Push(connection, source_file, device_filename,
                                mtime=int(mtime))
+    source_file.close()
```

Comment:
should only close it if we opened it. Can you add a flag on line 151 and close it if it's true?